docs/overview/deep_dive/formatting.rst

```diff
@@ -2,132 +2,189 @@ Formatting
+Common Issues with Pre-Commit
+-----------------------------
 
-.. code-block:: none
+As pre-commit hook runs before each commit, when it fails it provides an error message that's readable on terminals
+but not on IDE GUIs. So you might see a cryptic error message like one of the following:
 
-    black....................................................................Passed
-    flake8...................................................................Passed
+.. image:: https://github.com/unifyai/unifyai.github.io/blob/master/img/externally_linked/deep_dive/formatting/vscode_error.png?raw=true
+   :alt: git commit error in VS Code
+
+.. image:: https://github.com/unifyai/unifyai.github.io/blob/master/img/externally_linked/deep_dive/formatting/pycharm_error.png?raw=true
+   :alt: git commit error in PyCharm
+
+We recommend you commit your code from the terminal when you contribute to Ivy. But if you want to commit from your IDE,
+you can always either click on "Show Command Output" or "Show details in console" to see the error message.
+
+And be aware that some of the linters we use format your code automatically like ``black`` and ``autoflake``. So you
+will need to add the changes to your commit and commit again.
+
+Continuous Integration
+----------------------
+
+We have multiple GitHub actions to check and fix the formatting of the code. They can be divided into lint checks and
+lint formatting (or lint-bot).
+
+All the check we do are made by pre-commit, you don't need to worry about lint errors arising from the CI checks that
+are not caught by pre-commit.
+
+Lint Checks
+~~~~~~~~~~~
+
+We have a GitHub action that runs:
+
+1. Every commit
+2. Every pull request
+
+The important check is the one that runs on every pull request. You should expect this check to pass if you have
+pre-commit correctly set up. Note that you can also reformat your code directly from GitHub making a comment with
+``ivy-gardener``, we will go through this in the next section.
+
+Lint Formatting
+~~~~~~~~~~~~~~~
+
+We have a GitHub action that runs:
+
+1. Every day at 08:00 UTC
+2. Manually invoked by making a comment with ``ivy-gardener`` on a PR
 
+The first action is to ensure that the code is always formatted correctly. The second action is to allow you to
+reformat your code directly from GitHub. This is useful if you didn't setup pre-commit correctly and you or one
+of our maintainers want to reformat your code without having to clone the repository.
 
 **Round Up**
 
-This should have hopefully given you a good feel for how function wrapping is applied to functions in Ivy.
+This should have hopefully given you a good feel for what is our coding style and how to format your code to contribute
+to Ivy.
 
 If you have any questions, please feel free to reach out on `discord`_ in the `formatting channel`_ or in the `formatting forum`_!
 
+.. This video should be considered deprecated. It will be replaced with a new video soon.
```

Comment:
Let's use the new video

Reply:
Looks like I edited the video and forgot to delete the comment 😅